syntax = "proto3";

package instinct.data;

message OSSStatus {
  bool has_error = 1;
  string message = 2;
  ErrorType error_type = 3;
  enum ErrorType {
    UnknownOSSError = 0;
    ObjectNotFound = 1;
    FileNotOpen = 2;
  }
}

message Aggregations {
    message Aggregation {
      map<string, int64> int64 = 1;
      map<string, double> double = 2;
    }
    repeated Aggregation rows = 1;
}

//message PubObjectRequest {
//
//}
//
//message PutObjectResponse {
//
//}
//
//message GetObjectRequest {
//
//}
//
//message GetObjectResponse {
//
//}
//
//message DeleteObjectRequest {
//
//}
//
//message DeleteObjectResponse {
//
//}

